clumper/clump.py

```diff
@@ -62,6 +63,56 @@ def read_json(cls, path):
             return Clumper(data)
         return Clumper(json.loads(pathlib.Path(path).read_text()))
 
+    @classmethod
+    def read_csv(cls, path, delimiter=",", fieldnames=None, nrows=None):
+        """
+        Reads in a csv file. Can also read files from url.
+
+        Parameters
+        ----------
+        path : filename or url
+        delimiter: must be a single character. `,` is the default.
+        fieldnames: If `None`, the first row of the csv is assumed as the header.
+        nrows: Number of rows to read in. Useful when reading large files. If `None`, all rows are read.
+
+        Usage:
+
+        ```python
+        from clumper import Clumper
+
+        clump = Clumper.read_csv("tests/iris.csv")
+        assert len(clump) == 150
+
+        clump = Clumper.read_csv("tests/iris.csv", nrows = 30)
+        assert len(clump) == 30
+
+        clump = Clumper.read_csv("https://calmcode.io/datasets/monopoly.csv")
+        assert len(clump) == 22
+        ```
+        """
+        if path.startswith("https:") or path.startswith("http:"):
+            with urllib.request.urlopen(path) as resp:
+
+                if fieldnames is not None:
+                    fieldnames = fieldnames
+                else:
+                    fieldnames = resp.readline().decode().strip().split(",")
```

Comment:
This can be simplified. 

```python
if fieldnames is None: 
    fieldnames = resp.readline().decode().strip().split(",")
```

Since fieldnames is already set, there is no need for logic that handles `fieldnames = fieldnames`.